app/common/model/src/main/java/io/syndesis/common/model/integration/Flow.java

```diff
@@ -44,6 +43,23 @@
         // allow access to ImmutableIntegration.Builder
     }
 
+    enum MetadataKey {
+        TYPE;
+
+        public String value() {
+            return name().toLowerCase(Locale.US);
```

Comment:
Might be a bit dangerous to put `"type"` in metadata as it can easily be overwritten by code trying to store some other kind of _type_, say "async" or "reactive".
Do we need to store this in metadata, can we rather create enum like `FlowAxis` with values `PRMARY` and `ALTERNATE` and store it in property of the `Flow`?

and nit: not a big fan of doing computation over constant values, I'd cache this value in the constructor

Reply:
btw do we really need distinction between lower and upper case ?

Reply:
I like the idea of having this as a property on the flow. Let me try this ...

Reply:
maybe we can also move the meta key to the metadata itself and list all the know values

Reply:
and maybe the meta can also have a method that leverages Enum’s static method:

    public static <T extends Enum<T>> T valueOf(Class<T> enumType,
                            String name)